### PR TITLE
Fix outstanding Tensorflow and Numpy errors as mentioned in #1029

### PR DESCRIPTION
**Pull request by MuawizChaudhary**
As noted in #1029, the stack operation stacks frequency(?) coefficients on the wrong axis in numpy and tensorflow implementation of frequency scattering, and the Fourier sub-sampling absorbs specific dims that it shouldn't in Numpy and Tensorflow. 

This PR modifies the core time frequency algorithm to stack on the 2nd axis, modifies the Fourier sub-sampling to be agnostic in dim up to sub-sampling dim in numpy and tensorflow, and slightly modifies tests to account for the correct stacking behavior and to include tensorflow in more tests. 

@lostanlen @cyrusvahidi would appreciate your thoughts and review.

### CODE REVIEW

File: kymatio/scattering1d/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -27,9 +27,14 @@ def subsample_fourier(cls, x, k):
         """
         cls.complex_check(x)
 
-        y = tf.reshape(x, (-1, k, x.shape[-1] // k))
+        shape = list(x.shape[:-1]) + [k, x.shape[-1]//k]
+        if shape[0] is None:
+            shape[0] = -1
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/backend/tensorflow_backend.py:32]: @janden @eickenberg do y'all think this additional code is fine? 

The Keras backend was erroring out when I tried to reshape. x.shape produces a datatype with None as the first element. The None specifies a general "batch" shape in keras. I need to convert this None to a -1 to play nice with Tensorflow.

Comment (edouardoyallon) [reply kymatio/scattering1d/backend/tensorflow_backend.py:32]: This sounds good to me!

**Review by edouardoyallon — APPROVED** [review]

**Comment by edouardoyallon** [comment]
looks good to me (on the other hand my last PR was a failure ..)
